### PR TITLE
Improve instructions for better results and better code readability in ChatGPT tool

### PR DESCRIPTION
**Pull request by arash77**
This pull request improves the instructions for better results and better code readability. It updates the instructions for the assistant to provide more specific guidance on handling file search queries and image file queries. It also adds error handling for unsupported file types and models for vision. Additionally, the version suffix in the integration with OpenAI's ChatGPT into Galaxy has been incremented to 1.

### CODE REVIEW

File: tools/chatgpt/chatgpt.py [hunk 1]
```diff
@@ -39,34 +39,38 @@
...
 
 file_search_file_streams = []
-image_urls = []
+image_files = []
 
 for path in context_files:
     ext = path.split(".")[-1].lower()
-    if ext in vision_sup_ext and model in ["gpt-4o", "gpt-4o-mini", "gpt-4-turbo"]:
+    if ext in vision_sup_ext:
+        if model not in ["gpt-4o", "gpt-4o-mini", "gpt-4-turbo"]:
```
Comment (bgruening) [inline tools/chatgpt/chatgpt.py:47]: should we maybe instead, or in addition, prevent this error in the UI?

File: tools/chatgpt/chatgpt.py [hunk 2]
```diff
@@ -39,34 +39,38 @@
...
         file = client.files.create(file=open(path, "rb"), purpose="vision")
         promt = {"type": "image_file", "image_file": {"file_id": file.id}}
-        image_urls.append(promt)
-
+        image_files.append(promt)
     elif ext in file_search_sup_ext:
         file_search_file_streams.append(open(path, "rb"))
     else:
-        raise Exception("Not supported file!")
+        raise Exception(f"File type {ext} is not supported!")
```
Comment (bgruening) [inline tools/chatgpt/chatgpt.py:59]: why can this even happen? We can define which filetypes are allowed in the UI

Comment (arash77) [reply tools/chatgpt/chatgpt.py:60]: It never happens, it is just for caution.

File: tools/chatgpt/chatgpt.py [hunk 3]
```diff
@@ -4,6 +4,8 @@
 from openai import OpenAI
 
 context_files = sys.argv[1].split(",")
+print(context_files)
+sys.exit(0)
```
Comment (bgruening) [inline tools/chatgpt/chatgpt.py:8]: that is wired ... ;)

**Review by bgruening — COMMENTED** [review]
please alsway bump the version number.

File: tools/chatgpt/chatgpt.py (hunk 3 above)
Comment (arash77) [reply tools/chatgpt/chatgpt.py]: It was just for testing I forgot to remove it, sorry!

**Comment by bgruening** [comment]
The version suffix

**Comment by arash77** [comment]
> The version suffix

I have already done it here:
https://github.com/bgruening/galaxytools/pull/1486/files#diff-16baf79c2e3cb928410eb55211519c93777b1ab046be10aee889fa9f12e30548R5

File: tools/chatgpt/chatgpt.xml [hunk 4]
```diff
@@ -32,17 +34,30 @@ $__user__.extra_preferences.get('chatgpt|api_key', "")
...
+            <when value="vison">
+                <param name="model" type="select" optional="false" label="Model" help="Select the model you want to use">
+                    <option value="gpt-4o-mini" selected="true">Affordable and intelligent small model for fast, lightweight tasks (gpt-4o-mini)</option>
+                    <option value="gpt-4o">High-intelligence flagship model for complex, multi-step tasks (gpt-4o)</option>
+                    <option value="gpt-4-turbo">The previous set of high-intelligence model with vision capabilities (gpt-4-turbo)</option>
+                </param>
+                <param name="context" type="data" multiple="true" optional="false" format="doc,docx,html,json,pdf,txt,jpg,jpeg,png,webp,gif" label="Context" max="500" help="This data will be uploaded to OpenAI's servers for processing."/>
+            </when>
+            <when value="no_vision">
+                <param name="model" type="select" optional="false" label="Model" help="Select the model you want to use">
```
Comment (bgruening) [inline tools/chatgpt/chatgpt.xml:51]: All models are "no-vision" models, aren't they? So I would expect here all models

**Pull request merged**